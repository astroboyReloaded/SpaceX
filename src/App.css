.mainHeader {
  display: flex;
  padding-inline: 1.8rem;
  align-items: center;
  border-bottom: 1px solid gray;
}

.appLogo {
  height: 60px;
}

.appTitle {
  margin-inline: 1rem;
}

.navBar {
  margin-left: auto;
}

.navList {
  display: flex;
  list-style: none;
  margin: 1rem;
}

.navListItem:last-child {
  border-left: 1px solid gray;
}

a {
  padding: 1rem;
  text-decoration: none;
}

.active {
  text-decoration: underline;
}

main {
  padding: 1.8rem;
}