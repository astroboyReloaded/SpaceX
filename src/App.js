import { Routes, Route } from 'react-router-dom';
import './App.css';
import Nav from './components/Nav';

const App = () => (
  <Routes>
    <Route path="/" element={<Nav />}>
      <Route index  element={<h1>Rockets</h1>}/>
      <Route path="/missions" element={<h1>Missions</h1>}/>
      <Route path="/profile" element={<h1>My Profile</h1>}/>
      <Route path="*" element={<h1>Page Not Found</h1>}/>
    </Route>
  </Routes>
);

export default App;
