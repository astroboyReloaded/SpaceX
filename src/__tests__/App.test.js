import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Routes, Route} from 'react-router-dom';
import { Provider } from 'react-redux';
import Store from '../redux/Store';
import App from '../App';

it('Renders App', () => {
  const tree = renderer
    .create(
      <Router>
        <Provider store={Store}>
          <Routes>
            <Route path="*" element={<App />} />
          </Routes>
        </Provider>
      </Router>,
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
