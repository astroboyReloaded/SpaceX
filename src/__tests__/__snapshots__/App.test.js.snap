// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders App 1`] = `
Array [
  <header
    className="mainHeader"
  >
    <img
      alt="App logo"
      className="appLogo"
      src="logo.png"
    />
    <h1
      className="appTitle"
    >
      Space Traveer's Hub
    </h1>
    <nav
      className="navBar"
    >
      <ul
        className="navList"
      >
        <li
          className="navListItem"
        >
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Rockets
          </a>
        </li>
        <li
          className="navListItem"
        >
          <a
            className=""
            href="/missions"
            onClick={[Function]}
          >
            Missions
          </a>
        </li>
        <li
          className="navListItem"
        >
          <a
            className=""
            href="/profile"
            onClick={[Function]}
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main>
    <section />
  </main>,
]
`;
